src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -208,13 +206,35 @@ internal Cci.IFieldReference CreateDataField(ImmutableArray<byte> data, ushort a
 
             Cci.ITypeReference type = _proxyTypes.GetOrAdd(
                 ((uint)data.Length, Alignment: alignment), key =>
-                key.Size switch
                 {
-                    1 when _systemInt8Type is not null => _systemInt8Type,
-                    2 when _systemInt16Type is not null => _systemInt16Type,
-                    4 when _systemInt32Type is not null => _systemInt32Type,
-                    8 when _systemInt64Type is not null => _systemInt64Type,
-                    _ => new ExplicitSizeStruct(key.Size, key.Alignment, this, _systemValueType)
+                    // We need a type that's both the same size as the data and that has a .pack
+                    // that matches the data's alignment requirements.
+
+                    if (key.Size == 1)
+                    {
+                        // Data of size 1 must be a single element of one-byte-sized type,
+                        // which has no alignment requirements.
+                        Debug.Assert(alignment == 1);
```

Comment:
>alignment [](http://example.com/codeflow?start=37&length=9)

This looks like an unintentional capturing of the ```alignment``` parameter. Consider marking the lambda as static.

Reply:
It can't be static as it captures this.